From 8cc25723179f99a2d1f089dce8110a85799f8f8a Mon Sep 17 00:00:00 2001
From: Wang Yuanheng <wangyuanheng@wxiat.com>
Date: Wed, 20 Jul 2022 15:55:57 +0800
Subject: [PATCH] sw64: fix compile errors when CONFIG_KVM=m
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 8cc25723179f99a2d1f089dce8110a85799f8f8a
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

Export symbol for bind_vcpu_enabled to fix compile errors when
CONFIG_KVM=m.

Signed-off-by: Wang Yuanheng <wangyuanheng@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/bindvcpu.c | 3 ++-
 arch/sw_64/kvm/kvm-sw64.c    | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/sw_64/kernel/bindvcpu.c b/arch/sw_64/kernel/bindvcpu.c
index 611c395c144b..46617eb68b7a 100644
--- a/arch/sw_64/kernel/bindvcpu.c
+++ b/arch/sw_64/kernel/bindvcpu.c
@@ -11,7 +11,8 @@
 #include <linux/uaccess.h>
 #include <asm/debug.h>
 
-extern bool bind_vcpu_enabled;
+__read_mostly bool bind_vcpu_enabled;
+EXPORT_SYMBOL(bind_vcpu_enabled);
 
 static int __init bind_vcpu_init(void)
 {
diff --git a/arch/sw_64/kvm/kvm-sw64.c b/arch/sw_64/kvm/kvm-sw64.c
index 8ba7e18698b3..9d209141820c 100644
--- a/arch/sw_64/kvm/kvm-sw64.c
+++ b/arch/sw_64/kvm/kvm-sw64.c
@@ -22,7 +22,7 @@
 bool set_msi_flag;
 unsigned long sw64_kvm_last_vpn[NR_CPUS];
 #if defined(CONFIG_DEBUG_FS) && defined(CONFIG_NUMA)
-__read_mostly bool bind_vcpu_enabled;
+extern bool bind_vcpu_enabled;
 #endif
 #define cpu_last_vpn(cpuid) sw64_kvm_last_vpn[cpuid]
 
-- 
2.33.0

